rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -234,14 +237,14 @@ rmw_ret_t rmw_connextdds_extend_initial_peer_list(
          then free their memory. */
       DDS_String_free(*element_ref);
     }
-    const auto peer_char_len = strlen(peer);
-    *element_ref = DDS_String_alloc(peer_char_len);
+    const auto peer_char_len = strnlen(peer, RMW_DISCOVERY_OPTIONS_STATIC_PEERS_MAX_LENGTH);
+    *element_ref = DDS_String_alloc(peer_char_len + 1);  // +1 for nul character
```

Comment:
Minor, but `DDS_String_alloc` [already accounts for the nul terminator](https://community.rti.com/static/documentation/connext-dds/6.0.1/doc/api/connext_dds/api_c/group__DDSStringClass.html#gac243227428eb05799e9f1ab30aeefa85)